client/src/globalState.tsx

```diff
@@ -0,0 +1,274 @@
+
+  const [unreadFlags, setUnreadFlags] = useState<PerPinboard<boolean>>({});
+
+  const setUnreadFlag = (pinboardId: string) => (
+    unreadFlag: boolean | undefined
+  ) => {
+    setUnreadFlags((prevUnreadFlags) => ({
+      ...prevUnreadFlags,
+      [pinboardId]: unreadFlag,
+    }));
+    !unreadFlag && clearDesktopNotificationsForPinboardId(pinboardId);
+  };
+
+  const hasUnread =
+    Object.values(unreadFlags).find((unreadFlag) => unreadFlag) || false;
+
+  const hasUnreadOnOtherPinboard = (pinboardId: string): boolean =>
+    !!hasUnread &&
+    !!Object.entries(unreadFlags).find(
+      ([otherPinboardId, isUnread]) =>
+        isUnread && pinboardId !== otherPinboardId
+    );
+
+  const closePinboard = (pinboardIdToClose: string) => {
+    if (activePinboardIds.includes(pinboardIdToClose)) {
+      setManuallyOpenedPinboards([
+        ...manuallyOpenedPinboards.filter(
+          (pinboard) => pinboard.id !== pinboardIdToClose
+        ),
+      ]);
+    }
+    setSelectedPinboardId(null);
+    setUnreadFlag(pinboardIdToClose)(undefined);
+    setError(pinboardIdToClose, undefined);
+  };
+
+  useSubscription(gqlOnCreateItem(), {
+    onSubscriptionData: ({ subscriptionData }) => {
+      const { pinboardId, mentions } = subscriptionData.data.onCreateItem;
+
+      const isMentioned = mentions.includes(userEmail); // TODO also check group membership here (once added)
+
+      const pinboardIsOpen = isExpanded && selectedPinboardId === pinboardId;
+
+      if (isMentioned && !pinboardIsOpen) {
+        setUnreadFlag(pinboardId)(true);
+      }
+    },
+  });
+
+  useEffect(() => {
+    window.addEventListener("message", (event) => {
+      if (
+        event.source !== window &&
+        Object.prototype.hasOwnProperty.call(event.data, "item")
+      ) {
+        const item = event.data.item;
+        window.focus();
+        setIsExpanded(true);
+        setSelectedPinboardId(item.pinboardId); // FIXME handle if said pinboard is not active (i.e. load data)
+        // TODO ideally highlight the item
+      }
+    });
+  }, []);
+
+  const isNotTrackedInWorkflow = Boolean(
+    preselectedComposerId &&
+      !preselectedPinboard &&
+      !preselectedPinboardQuery.loading
+  );
+
+  const contextValue = {
+    userEmail,
+    userLookup,
+
+    activePinboardIds,
+    payloadToBeSent,
+    clearPayloadToBeSent,
+
+    openPinboard,
+    closePinboard,
+    preselectedPinboard,
+    clearSelectedPinboard,
+
+    showNotification,
+    hasWebPushSubscription,
+
+    errors,
+    setError,
+    hasErrorOnOtherPinboard,
+
+    unreadFlags,
+    setUnreadFlag,
+    hasUnreadOnOtherPinboard,
+
+    selectedPinboardId,
+    isExpanded,
+    setIsExpanded,
+  };
+
+  return (
+    <GlobalStateContextProvider value={contextValue}>
+      <Floaty
+        presetUnreadNotifications={presetUnreadNotifications}
+        isExpanded={isExpanded}
+        setIsExpanded={setIsExpanded}
+        hasError={hasError}
+        hasUnread={hasUnread}
```

Comment:
if we're exposing so much via the context, is there any need to pass these as props?

Reply:
if we do it all via the context, we can avoid `GlobalState` having to combine the Floaty and Panel, and instead just export the Provider, then combine them in `app.tsx` - avoiding the need for the gross name in  https://github.com/guardian/editorial-tools-pinboard/pull/81#discussion_r816869493